DSP/APCompressor.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+  ==============================================================================
+
+    APCompressor.cpp
+    Created: 26 Feb 2021 2:08:02pm
+    Author:  Johnathan Handy
+
+  ==============================================================================
+*/
+
+#include "APCompressor.h"
+#include "JuceHeader.h"
+
+APCompressor::APCompressor()
```

Comment:
The compression functions in this class have a lot in common. See if you can refactor them to reduce repeated code (DRY principle)